dist/prow_yaml_lint.sh

```diff
@@ -2,17 +2,6 @@
 
 set -euxo pipefail
 
-YAML_LINTER='yamllint'
-YAML_RULES='{extends: default, rules: {line-length: {max: 120}}}'
-YAML_LINT_CMD=("${YAML_LINTER}" '-s' '-d' "${YAML_RULES}")
+yamllint -v
 
-if ! type -f "${YAML_LINTER}"; then
-  echo "error: could not find ${YAML_LINTER} in PATH"
-  exit 1
-fi
-
-find . \
-  -path "./cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/test_fixtures" -prune -o \
-  -path "./vendor" -prune -o \
-  -path "./dist/grafana/dashboards" -prune -o \
-  -type f \( -name '*.yaml' -o -name '*.yml' \) -print0 | xargs -L 1 -0 "${YAML_LINT_CMD[@]}"
+yamllint --list-files -s -d '{extends: default, rules: {line-length: {max: 120}}, ignore: [cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/test_fixtures/, vendor/, dist/grafana/dashboards/]}' .
```

Comment:
TBH I am not a big fan of oneliners like that, they are hard to read and understand if you are not familiar with the tool (what does `-d` do? What does `--list-files -s` do?)

Could we keep the rules and ignored directories in separate variables and assemble the -d parameter value dynamically from that?

Reply:
`--list-file` is explained in the PR description (and now removed <--- explained later).
`-s` and `-d` were there before this PR.

```console
$ yamllint -h | rg 'strict|config-data|list-f'
usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA] [--list-files]
  -d CONFIG_DATA, --config-data CONFIG_DATA
  --list-files          list files to lint and exit
  -s, --strict          return non-zero exit code on warnings as well as
```

For the readability, I would go with a `.yamllint.yaml` file in the repo. WDYT?
(I did not go there directly to keep my change minimal, basically reformatting several commands into one.)

Just found that I need to remove `--list-files` 😛 as it does not do lint at all.

Reply:
I understand it in the context of this PR, but will not be happy to see the oneliner in the repo if I need to change or understand it. Previous script was a bit verbose, yes, but the structure helped understanding the meaning - what are the rules, and what paths are excluded. That is now all encoded in an inline JSON, which I dislike.

Reply:
Could we structure the config YAML  (`.yamllint` file sounds a bit like overkill).

```
cfg=$(mktemp)
cat >$cfg <<EOF
extends: default
rules:
  ...
ignore:
- path
EOF

yamllint -c $cfg
```

Reply:
We lose the chance to yamlint on the inline config file. :) 
But it definitely gets better than the config data arg. in the command.